### PR TITLE
[SwiftLint] Disabling Some Rules

### PR DESCRIPTION
**Pull request by adamrushy**
### 🎯 Goal

This brings us to 0 warnings from SwiftLint

### 🛠 Implementation

There is some rules disabled for some larger functions that will require bigger refactors.

### 🧪 Testing

CMD + B

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: DemoApp/CustomMessageContentView.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ class CustomMessageContentView: ChatMessageContentView {
             }
         }
         
-        guard let authorNameLabel = authorNameLabel, authorNameLabel.text != "" else {
+        guard let authorNameLabel = authorNameLabel, let authorText = authorNameLabel.text, authorText.isEmpty else {
```
Comment (nuno-vieira) [inline DemoApp/CustomMessageContentView.swift:25]: This also works and is a bit shorter and easier to read 👍 
```suggestion
        guard let authorNameLabel = authorNameLabel, authorNameLabel.text?.isEmpty == false else {
```

File: StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -1,61 +1,59 @@
...
-        }
-      },
-      {
-        "package": "Swifter",
-        "repositoryURL": "https://github.com/httpswift/swifter.git",
-        "state": {
-          "branch": null,
-          "revision": "9483a5d459b45c3ffd059f7b55f9638e268632fd",
-          "version": "1.5.0"
-        }
```
Comment (nuno-vieira) [inline StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:56]: Why are we updating the dependencies? 🤔

File: .swiftlint.yml [hunk 3]
```diff
@@ -10,6 +10,9 @@ excluded:
   - vendor
   - UISDKdocumentation
   - Tests
+  - Sources/StreamChatUI/Deprecations.swift
+  - Sources/StreamChatUI/**/**/AttachmentViewCatalog.swift
+  - Sources/StreamChat/Controllers/DataController.swift
```
[the remark this answers is not in the document]
Comment (bielikb) [reply .swiftlint.yml:15]: Lets not exclude the source files on this level.

File: DemoApp/CustomMessageContentView.swift (hunk 1 above)
Comment (adamrushy) [reply DemoApp/CustomMessageContentView.swift:25]: Yeah, I considered this one but I preferred the readability of the one I implemented, wdyt?

File: .swiftlint.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamrushy) [reply .swiftlint.yml:15]: These require bigger refactors, so I will be spinning up tickets to cover these (unless we want the warnings to exist during this time)

[the remark this answers is not in the document]
Comment (bielikb) [reply .swiftlint.yml:15]: ok. please create those tickets. Apart from this point, all looks good == +1 from me.

File: StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 2 above)
Comment (bielikb) [reply StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:56]: json is unordered dictionary. I believe what happened here is that the `Package.resolved` got regenerated.

**Review by bielikb — APPROVED** [review]

File: StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 2 above)
Comment (nuno-vieira) [reply StreamChat.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:56]: The new JSON has Atlantis 1.16.0, so it looks like we updated Atlantis from 1.15.0 -> 1.16.0. But I think it is fine though 👍

**Pull request merged**